test/fuzz/lua/test_engine.lua

```diff
@@ -0,0 +1,1411 @@
+    space:insert(tuple)
+end
+
+local function upsert_op(space, tuple, tuple_ops)
+    assert(next(tuple_ops) ~= nil)
+    space:upsert(tuple, tuple_ops)
+end
+
+local function update_op(space, key, tuple_ops)
+    assert(next(tuple_ops) ~= nil)
+    space:update(key, tuple_ops)
+end
+
+local function replace_op(space, tuple)
+    space:replace(tuple)
+end
+
+local function bsize_op(space)
+    space:bsize()
+end
+
+local function len_op(space)
+    space:len()
+end
+
+local function format_op(space, space_format)
+    space:format(space_format)
+end
+
+local function setup(engine_name, space_id_func, test_dir, verbose)
+    log.info('SETUP')
+    assert(engine_name == 'memtx' or
+           engine_name == 'vinyl')
+    -- Configuration reference (box.cfg),
+    -- https://www.tarantool.io/en/doc/latest/reference/configuration/
+    local box_cfg_options = {
+        checkpoint_count = math.random(5),
+        checkpoint_interval = math.random(60),
+        checkpoint_wal_threshold = math.random(1024),
+        iproto_threads = math.random(1, 10),
+        memtx_allocator = oneof({'system', 'small'}),
+        memtx_memory = 1024 * 1024,
+        memtx_sort_threads = math.random(1, 256),
+        memtx_use_mvcc_engine = oneof({true, false}),
+        readahead = 16320,
+        slab_alloc_factor = math.random(1, 2),
+        vinyl_bloom_fpr = math.random(50) / 100,
+        vinyl_cache = oneof({0, 2}) * 1024 * 1024,
+        vinyl_max_tuple_size = math.random(0, 100000),
+        vinyl_memory = 800 * 1024 * 1024,
+        vinyl_page_size = math.random(1024, 2048),
+        vinyl_range_size = 128 * 1024,
+        vinyl_read_threads = math.random(2, 10),
+        vinyl_run_count_per_level = math.random(1, 10),
+        vinyl_run_size_ratio = math.random(2, 5),
+        vinyl_timeout = math.random(1, 5),
+        vinyl_write_threads = math.random(2, 10),
+        wal_cleanup_delay = 14400,
+        wal_dir_rescan_delay = math.random(1, 20),
+        wal_max_size = math.random(1024 * 1024 * 1024),
+        wal_mode = oneof({'write', 'fsync'}),
+        wal_queue_max_size = 16777216,
+        work_dir = test_dir,
+        worker_pool_threads = math.random(1, 10),
+    }
+    if verbose then
+        box_cfg_options.log_level = 'verbose'
+    end
+    box.cfg(box_cfg_options)
+    log.info('FINISH BOX.CFG')
+
+    log.info('CREATE A SPACE')
+    local space_format = random_space_format()
+    -- TODO: support `constraint`.
+    -- TODO: support `foreign_key`.
+    local space_opts = {
+        engine = engine_name,
+        field_count = oneof({0, table.getn(space_format)}),
+        format = space_format,
+        if_not_exists = oneof({true, false}),
+        is_local = oneof({true, false}),
+    }
+    if space_opts.engine ~= 'vinyl' then
+        space_opts.temporary = oneof({true, false})
+    end
+    local space_name = ('test_%d'):format(space_id_func())
+    local space = box.schema.space.create(space_name, space_opts)
```

Comment:
Let's create a few test spaces (say 3-4), not just one.

Reply:
I would postpone this and make the test script more complicated when current version will stop to find new bugs. Sure, there is a room for improvements in the test script, but we should put it to a master as soon as possible to make it available for everyone.